Tests/ShamTests/Tests/StubRequestTests.swift

```diff
@@ -25,4 +25,32 @@ final class StubRequestTests: XCTestCase {
         let request = StubRequest(url: self.baseURLString)
         XCTAssertEqual(request.description, "ALL: \(self.baseURLString)")
     }
+    
+    // MARK: Query items tests
+    
+    func testAccuratelyValidatesWhenARequestHasAllQueryItems() {
```

Comment:
I'm a little confused by the name of this test and the assertion at the end. The assertion passes when one StubRequest has a subset of query parameters from another StubRequest, and `.allowMissingQueryParameters` is used. But the test name (`ARequestHasAllQueryItem`) makes it seem like the two URLs would be exact matches.